src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -819,6 +819,19 @@ private void pushFightLoopOnStack(final boolean firstRun) {
 
       @Override
       public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        if (Match.someMatch(m_attackingUnits, Matches.unitHasAttackValueOfAtLeast(1))) {
```

Comment:
Are you sure this would cover all cases? Would it work if I had 2 units that provided negative support to each other therefore making them have a total of 0 attack? Would it work if enemy units provided negative support canceling out my attack value? 

I think in general, we can't be checking individual units when it comes to attack/defense values do to how flexible support attachments are (some maps do use negative support and/or supporting enemy units).